### PR TITLE
Release 2.15.1

### PR DESCRIPTION
**Pull request by pallavisontakke**
Release 2.15.1

This release contains performance improvements and bug fixes since
the 2.15.0 release. We recommend that you upgrade at the next available opportunity.

**Bugfixes**
* #6540 Segmentation fault when backfilling data with COPY into a compressed chunk
* #6858 Before update trigger not working correctly
* #6908 Fix gapfill with timezone behaviour around dst switches
* #6911 Fix dropped chunk metadata removal in update script
* #6940 Fix `pg_upgrade` failure by removing `regprocedure` from catalog table
* #6957 Fix segfault in UNION queries with ordering on compressed chunks

**Thanks**
* @DiAifU, @kiddhombre and @intermittentnrg for reporting issues with gapfill and daylight saving time
* @edgarzamora for reporting issue with update triggers
* @hongquan for reporting an issue with the update script
* @iliastsa and @SystemParadox for reporting an issue with COPY into a compressed chunk

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,26 @@
 `psql` with the `-X` flag to prevent any `.psqlrc` commands from
 accidentally triggering the load of a previous DB version.**
 
+## 2.15.1 (2024-05-28)
+
+This release contains performance improvements and bug fixes since
+the 2.15.0 release. We recommend that you upgrade at the next
+available opportunity.
```
Comment (svenklemm) [inline CHANGELOG.md:11]: Should probably wait for any wording changes by @billy-the-fish

**Review by svenklemm — APPROVED** [review]

**Comment by adschmu** [comment]
Is the 2.15.X-fix_hypertable_foreign_keys.sql script still required? It is not entirely clear whether "For [...] this release only" refers to 2.15.0 or 2.15.X.

The file name hints at it, but it would be helpful to include the one sentence about the script into these release notes again, if it still applies. There might be people not reading all of the changelogs, and the script is more like a "Known issues" bullet point anyway ...

**Review by iroussos — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,26 @@
...
+
+**Bugfixes**
+* #6540 Segmentation fault when backfilling data with COPY into a compressed chunk
```
Comment (billy-the-fish) [inline CHANGELOG.md:14]:
```suggestion
* #6540: Segmentation fault when you backfill data using COPY into a compressed chunk.
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,6 +4,26 @@
...
+* #6858 Before update trigger not working correctly
```
Comment (billy-the-fish) [inline CHANGELOG.md:15]:
```suggestion
* #6858: `BEFORE UPDATE` trigger not working correctly.
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -4,6 +4,26 @@
...
+* #6908 Fix gapfill with timezone behaviour around dst switches
```
Comment (billy-the-fish) [inline CHANGELOG.md:16]:
```suggestion
* #6908: Fix `time_bucket_gapfill()` with timezone behaviour around daylight savings time (DST) switches. 
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -4,6 +4,26 @@
...
+* #6911 Fix dropped chunk metadata removal in update script
```
Comment (billy-the-fish) [inline CHANGELOG.md:17]:
```suggestion
* #6911: Fix dropped chunk metadata removal in the update script.
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -4,6 +4,26 @@
...
+* #6940 Fix `pg_upgrade` failure by removing `regprocedure` from catalog table
```
Comment (billy-the-fish) [inline CHANGELOG.md:18]:
```suggestion
* #6940: Fix `pg_upgrade` failure by removing `regprocedure` from the catalog table.
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -4,6 +4,26 @@
...
+* #6957 Fix segfault in UNION queries with ordering on compressed chunks
```
Comment (billy-the-fish) [inline CHANGELOG.md:19]:
```suggestion
* #6957: Fix the `segfault` in UNION queries that contain ordering on compressed chunks.
```

File: CHANGELOG.md [hunk 8]
```diff
@@ -4,6 +4,26 @@
...
+
+**Thanks**
+* @DiAifU, @kiddhombre and @intermittentnrg for reporting issues with gapfill and daylight saving time
```
Comment (billy-the-fish) [inline CHANGELOG.md:22]:
```suggestion
* @DiAifU, @kiddhombre and @intermittentnrg for reporting the issues with gapfill and daylight saving time.
```
```suggestion
* @DiAifU, @kiddhombre and @intermittentnrg for reporting issues with gapfill and daylight saving time
```

File: CHANGELOG.md [hunk 9]
```diff
@@ -4,6 +4,26 @@
...
+* @edgarzamora for reporting issue with update triggers
```
Comment (billy-the-fish) [inline CHANGELOG.md:23]:
```suggestion
* @edgarzamora for reporting the issue with update triggers.
```

File: CHANGELOG.md [hunk 10]
```diff
@@ -4,6 +4,26 @@
...
+* @hongquan for reporting an issue with the update script
```
Comment (billy-the-fish) [inline CHANGELOG.md:24]:
```suggestion
* @hongquan for reporting thge issue with the update script.
```

File: CHANGELOG.md [hunk 11]
```diff
@@ -4,6 +4,26 @@
...
+* @iliastsa and @SystemParadox for reporting an issue with COPY into a compressed chunk
```
Comment (billy-the-fish) [inline CHANGELOG.md:25]:
```suggestion
* @iliastsa and @SystemParadox for reporting the issue with COPY into a compressed chunk.
```

**Review by billy-the-fish — CHANGES_REQUESTED** [review]
Have added a few suggestions. Happy follow up.

**Comment by svenklemm** [comment]
> Is the 2.15.X-fix_hypertable_foreign_keys.sql script still required? It is not entirely clear whether "For [...] this release only" refers to 2.15.0 or 2.15.X.
> 
> The file name hints at it, but it would be helpful to include the one sentence about the script into these release notes again, if it still applies. There might be people not reading all of the previous changelogs, and the script is more like a "Known issues" bullet point anyway ...

If you upgraded to 2.15.0 and have run the script after that upgrade you dont need to rerun. But if you are coming from a version before 2.15.0 you still need to execute the script.

**Comment by adschmu** [comment]
> > Is the 2.15.X-fix_hypertable_foreign_keys.sql script still required? It is not entirely clear whether "For [...] this release only" refers to 2.15.0 or 2.15.X.
> > The file name hints at it, but it would be helpful to include the one sentence about the script into these release notes again, if it still applies. There might be people not reading all of the previous changelogs, and the script is more like a "Known issues" bullet point anyway ...
> 
> If you upgraded to 2.15.0 and have run the script after that upgrade you dont need to rerun. But if you are coming from a version before 2.15.0 you still need to execute the script.

Still, it would be nice to add this fact to the release notes for people coming from < 2.15.0, since not everybody might read the previous notes and IMHO it does not hurt much to include these two lines for 2.15.x minor releases:

```
**For on-premise users and upgrades from < 2.15.0 only**, you will need to run [this SQL script](https://github.com/timescale/timescaledb-extras/blob/master/utils/2.15.X-fix_hypertable_foreign_keys.sql) after running `ALTER EXTENSION`. More details can be found in the pull request [#6786](https://github.com/timescale/timescaledb/pull/6797).
```

**Comment by adschmu** [comment]
BTW the displayed number for PR in the SQL Script line is wrong, it should say 6797 instead of 6786 as in the link, which is correct. Maybe this can corrected for the 2.15.0 info as part of this commit.

File: CHANGELOG.md (hunk 1 above)
Comment (pallavisontakke) [reply CHANGELOG.md]: right.

**Comment by pallavisontakke** [comment]
> Have added a few suggestions. Happy follow up.

Edited so.

**Review by billy-the-fish — APPROVED** [review]
LGTM. Thank you.

**Pull request merged**